### PR TITLE
Fix the Paypal Express Model; Courier name and checkout_method

### PR DESCRIPTION
**Pull request by boesbo**
### Description (*)
The PR solves the problem of the shipping method name shown on the PayPal Express Checkout page (see screen below), and the incorrect management of the "checkout_method" in the event of a checkout that has started and is not completed, and then opts for the PayPal express payment from the cart.

I have been using this fix for about 2 years without problems.

#### Before:
before

#### After:
after

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Review by matteotestoni — DISMISSED** [review]

**Comment by fballiano** [comment]
I've rebased it to the "main" branch in the meanwhile

**Comment by boesbo** [comment]
News about?

File: app/code/core/Mage/Paypal/Model/Express/Checkout.php [hunk 1]
```diff
@@ -694,14 +694,14 @@ public function getCheckoutMethod()
         if ($this->getCustomerSession()->isLoggedIn()) {
             return Mage_Checkout_Model_Type_Onepage::METHOD_CUSTOMER;
         }
-        if (!$this->_quote->getCheckoutMethod()) {
```
Comment (kiatng) [inline app/code/core/Mage/Paypal/Model/Express/Checkout.php:697]: We do not have enough reviewers. I do not use paypal and so should refrain from reviewing. Also, I am not any good with the language in the screenshot. Having said that, it's not clear to me why the original code needs to change. 

If a store allows both guest and register in onepage, the changed code always return guest, which is not the same as the original code.

File: app/code/core/Mage/Paypal/Model/Express/Checkout.php [hunk 2]
```diff
@@ -815,8 +815,8 @@ protected function _prepareShippingOptions(
 
                 $options[$i] = new Varien_Object(array(
                     'is_default' => $isDefault,
-                    'name'       => trim("{$rate->getCarrier()} - {$rate->getMethodTitle()}", ' -'),
-                    'code'       => $rate->getCode(),
+                    'name'       => $rate->getCode(),
+                    'code'       => trim("{$rate->getCarrierTitle()} - {$rate->getMethodTitle()}", ' -'),
```
[the remark this answers is not in the document]
Comment (kiatng) [reply app/code/core/Mage/Paypal/Model/Express/Checkout.php:815]: Is this change valid for all the rates?

File: app/code/core/Mage/Paypal/Model/Express/Checkout.php (hunk 1 above)
Comment (boesbo) [reply app/code/core/Mage/Paypal/Model/Express/Checkout.php:697]: Commits solve 2 problems.
The first is historical Magento 1, where the PayPal Express Checkout page does not display the name of the shipment but the code, so the customer sees something unknown like "tablerate_bestway".

The second one doesn't change the checkout logic at all but removes the deprecated method getCheckoutMethod (@deprecated after 1.4 beta1 it is checkout module responsibility)

File: app/code/core/Mage/Paypal/Model/Express/Checkout.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (fballiano) [reply app/code/core/Mage/Paypal/Model/Express/Checkout.php:815]: to me this change feels too weird to be added in the core, "code => getName()" feels wrong, there has to be somewhere where this data gets filled incorrectly and IMHO we should look for that

**Pull request closed without merging**